enterprise/server/githubapp/githubapp.go

```diff
@@ -414,10 +416,45 @@ func (a *GitHubApp) maybeTriggerBuildBuddyWorkflow(ctx context.Context, eventTyp
 	if err != nil {
 		return err
 	}
+	if err := a.MaybeReindexRepo(ctx, row.GitRepository, wd, repoURL, tok.GetToken()); err != nil {
+		log.Debugf("Not indexing repo: %s", err)
+	}
 	return a.env.GetWorkflowService().HandleRepositoryEvent(
 		ctx, row.GitRepository, wd, tok.GetToken())
 }
 
+func (a *GitHubApp) MaybeReindexRepo(ctx context.Context, repo *tables.GitRepository, wd *interfaces.WebhookData, repoURL *gitutil.RepoURL, accessToken string) error {
+	codesearchService := a.env.GetCodesearchService()
+	if codesearchService == nil {
+		return nil
+	}
+	if wd.PushedBranch != "main" && wd.PushedBranch != "master" {
```

Comment:
nit: we get the default branch name in the webhook payload - also probably good to check that this is a push event

```suggestion
	if wd.EventName != webhook_data.EventName.Push || wd.PushedBranch != wd.TargetRepoDefaultBranch {
```